src/Chapter3.hs

```diff
@@ -1075,18 +1075,20 @@ class Append a where
 
 instance Append [a] where
     append :: [a] -> [a] -> [a]
-    append x y = x ++ y
+    append = (++)
 
 newtype Gold = Gold Int
 
 instance Append Gold where
     append :: Gold -> Gold -> Gold
-    append (Gold x) (Gold y) = Gold (x+y)
+    append (Gold x) (Gold y) = Gold (x + y)
 
--- didn't quite understand what is required in this task :/
---instance Append (Maybe a) where
---    append :: (Maybe a) -> (Maybe a) -> (Maybe a)
---    append (Just x) (Just y) = Just (append x y)
+instance Append (Maybe a) where
+    append :: (Maybe a) -> (Maybe a) -> (Maybe a)
+    append (Just x) (Just y) = append (Just x) (Just y)
+    append (Nothing) (Just y) = Just y
+    append (Just x) (Nothing) = Just x
+    append (Nothing) (Nothing) = Nothing
```

Comment:
This is an absolutely correct solution!

However, we can notice that these 3 cases could be reduced to 2!
Let's see what we can unite in here.

If some of the arguments are `Nothing`, you actually always return the second one! So you need to inspect only two cases: `Nothing` in the first position, and `Nothing` on the second one 🙂

Reply:
@vrom911, hi! Thank you for the comments. But here, if I remove 
`append (Nothing) (Nothing) = Nothing`
then when I try to call
`append Nothing Nothing`
GHCi says: "Non-exhaustive patterns in function append". Or did you mean something else when you said that these 3 cases could be reduced to 2?

Reply:
Oh, I see.
Yes, in this case it is not enough to remove the last case, but the previous two should be patched a bit. Specifically, to show that not only `Just` would work this way with `Nothing` but anything:

```haskell
    append Nothing y = y
    append x Nothing = x
```

Then it would be the same as the correct solution that you wrote in the first place 👍🏼